diff --git a/tsconsensus/bolt_store.go b/tsconsensus/bolt_store.go
index ca347cfc0..ec17b6766 100644
--- a/tsconsensus/bolt_store.go
+++ b/tsconsensus/bolt_store.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & AUTHORS
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build !loong64
+//go:build !loong64 !riscv64
 
 package tsconsensus
 
diff --git a/tsconsensus/bolt_store_no_bolt.go b/tsconsensus/bolt_store_no_bolt.go
index 33b3bd6c7..43b6222cc 100644
--- a/tsconsensus/bolt_store_no_bolt.go
+++ b/tsconsensus/bolt_store_no_bolt.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & AUTHORS
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build loong64
+//go:build loong64 riscv64
 
 package tsconsensus
 
